### PR TITLE
wasm: Add native support for walk() built-in and relations

### PR DESCRIPTION
**Pull request by tsandall**
This commit adds native support for the walk() built-in inside the C
library. As part of this change, the planner has been updated to treat
relational built-ins differently from normal built-in
functions. Specifically, the planner assumes the relation is
implemented by a function that generates the full extent of the
relation. The planner then generates a scan across the relation
output, unifying each value with the output operand.

Fixe #2771

### CODE REVIEW

**Review by patrick-east — COMMENTED** [review]
LGTM

**Review by patrick-east — DISMISSED** [review]

File: wasm/src/value.c [hunk 1]
```diff
@@ -832,6 +832,40 @@ opa_value *opa_value_shallow_copy(opa_value *node)
...
+        __opa_value_transitive_closure(result, cpy, child);
+        prev = curr;
+    }
+}
+
+opa_value *opa_value_transitive_closure(opa_value *v)
+{
+    opa_array_t *result = opa_cast_array(opa_array());
+    opa_array_t *path = opa_cast_array(opa_array());
+    __opa_value_transitive_closure(result, path, v);
```
Comment (koponen-styra) [inline wasm/src/value.c:864]: path should be freed?

Comment (tsandall) [reply wasm/src/value.c:864]: The result is a set of tuples containing the path and the value referred to by that path--by convention, the input is included in the output w/ an empty path, so we can't free it here.

Comment (koponen-styra) [reply wasm/src/value.c:864]: ah, right, missed the inclusion in the beginning of the __opa_value_transitive_closure.

**Review by koponen-styra — APPROVED** [review]
LGTM

**Review by patrick-east — APPROVED** [review]

**Pull request merged**